packages/wonder-blocks-button/src/components/button-core.tsx

```diff
@@ -331,170 +329,148 @@ export const _generateStyles = (
+                ...(kind === "primary"
+                    ? {
+                          outline: `${borderWidthKind.press} solid ${themeVariant.press.border}`,
+                          outlineOffset: outlineOffsetKind,
+                      }
+                    : undefined),
+                // secondary
+                ...(kind === "secondary"
+                    ? {
+                          borderColor: themeVariant.press.border,
+                          boxShadow: `inset 0 0 0 ${borderWidthKind.press} ${themeVariant.press.border}`,
+                      }
+                    : undefined),
+
+                // tertiary-specific styles
+                ...(kind === "tertiary"
+                    ? {
+                          textDecoration: "underline",
+                          textUnderlineOffset: theme.root.font.offset.default,
+                          textDecorationThickness:
+                              theme.root.sizing.underline.press,
+                      }
+                    : undefined),
             },
-            iconWrapperHovered: {
-                backgroundColor: themeColorAction.hover.icon,
-                color: themeColorAction.hover.foreground,
-            },
-        };
-    } else if (kind === "tertiary") {
-        const themeColorAction = theme.color.tertiary[actionType];
-
-        const focusStyling = {
-            outlineStyle: "solid",
-            borderColor: "transparent",
-            outlineColor: themeColorAction.focus.border,
-            outlineWidth: theme.border.width.focused,
-            borderRadius: theme.border.radius.default,
-        };
-        const activePressedStyling = {
-            color: themeColorAction.press.foreground,
-            textDecoration: "underline",
-            textDecorationThickness: theme.size.underline.active,
-            textUnderlineOffset: theme.font.offset.default,
-        };
-
-        const sharedDisabledStyling = {
-            ...disabledStateStyles,
-            textDecoration: "none",
-            textDecorationThickness: "unset",
-            textUnderlineOffset: "unset",
-            outline: "none",
-        };
-
-        newStyles = {
-            default: {
-                background: themeColorAction.default.background,
-                color: themeColorAction.default.foreground,
-                paddingInline: 0,
-                ":hover": {
-                    textUnderlineOffset: theme.font.offset.default,
-                    textDecoration: "underline",
-                    textDecorationThickness: theme.size.underline.hover,
-                },
-                ":focus-visible": focusStyling,
-                ":active": activePressedStyling,
+        },
+        disabled: {
+            cursor: "not-allowed",
+            ...disabledStatesStyles,
+            // NOTE: Even that browsers recommend to specify pseudo-classes in
+            // this order: link, visited, focus, hover, active, we need to
```

Comment:
I think there seems to be a mismatch between the comment and the order of styles (comment says: `focus, hover, active`, and the order is `hover, active, focus`. Let me know if I am misunderstanding something though!

Reply:
Good point, changing it!